Add text content to the Design System

dashouse opened the pull request:
This PR takes all of the content with a "READY" status and adds it to the design system.

I have also updated the `toc.njk` file to reveal all of the content that's currently in the Design System - this is temporary while the navigation is being built on another branch by @hannalaakso .

I haven't added any examples, as this work will be done after @alex-ju's work on the example / code blocks.

36degrees commented on src/components/checkboxes/index.md.njk:
```diff
@@ -0,0 +1,35 @@
+---
+title: Checkboxes
+description: Let users select one or more options by using the Checkboxes component.
+section: Components
```
I think a lot of this metadata is redundant, at least for now. No harm in keeping aliases for now I guess, but `section`, `theme`, `status`, `department` and `copy` should all be removed.

dashouse replied:
Fixed by 488acc1

dashouse replied:
@36degrees Looks like we are using section.

```
<div class="app-c-content__header">
     <p class="app-c-content__section-heading">{{ section }}</p>
     <h1 class="app-c-content__heading heading-xlarge">{{ title }}</h1>
</div>
```
Should I put it back in or should this be coming from the file structure?

dashouse replied:
@36degrees Just putting it back for now

36degrees commented on src/components/checkboxes/index.md.njk:
```diff
@@ -0,0 +1,35 @@
+
+## When not to use this component
+
+Don’t use the Checkboxes component if users can only choose one option from a selection. In this case, use the [Radios component](Insert URL to Radios component page).
```
Can we put the URLs for these in now?

dashouse replied:
Not yet - Need to confirm URL structure for patterns

36degrees commented on src/components/checkboxes/index.md.njk:
```diff
@@ -0,0 +1,35 @@
+
+## Research on this component
+
+Read a blog post about updates to the [Radios and Checkboxes components](https://designnotes.blog.gov.uk/2016/11/30/weve-updated-the-radios-and-checkboxes-on-gov-uk/).
```
Missing opening `[` for the link.

36degrees replied:
Ignore me, I can't read. However, ideally our links should be more descriptive than this – users of assistive technologies iterating through the links on the page will just see 'Radios and checkboxes components' which is not a good description for where it will take them /cc @amyhupe

amyhupe replied:
I agree. Can we put the opening [ before the word 'updates' please?

alex-ju replied:
Updated

dashouse commented:
Just committed a bunch of changes.

- Added `section` back in
- Added `theme` in patterns
- Added an if statement in the layout to show `theme` if there is one
- Removed the text-transform on headings (as this will make theme's Title Case)

alex-ju commented:
Rebased and updated.

hannalaakso commented on src/components/textarea/index.md.njk:
```diff
@@ -0,0 +1,26 @@
+---
```
This seems to contain the content for Tag?

alex-ju commented:
would've been nice to have the About page too (even if blank, better that 404) and the side nav dependent on main one. but I guess I'll make a separate PR for this.

36degrees commented:
The about page is the homepage, the link just needs updating to '/' – which I've done in #89

alex-ju commented:
@36degrees good catch, but I think we need to update the about/homepage link on the mobile navigation too ('cause we still have the 404 there)
